web/src/components/JobVisOptions.vue

```diff
@@ -0,0 +1,146 @@
+<template>
+  <v-list-item-content
+    :set="dataset = linkedDataset(pin)"
+  >
+    <v-list-item-title>{{ pinDisplayTitle(pin) }}</v-list-item-title>
+    <v-list-item-subtitle
+      v-for="info in additionalInfo(dataset)"
+      :key="info"
+    >
+      {{ info }}
+    </v-list-item-subtitle>
+    <div
+      v-for="visOption in visOptions(dataset)"
+      v-show="active"
+      :key="visOption.value"
+      :set="showOptions = visualizationsShown.map(vis => vis.value).includes(visOption.value)"
+      @click="(event) => event.stopPropagation()"
+    >
+      <v-checkbox
+        :value="showOptions"
+        :label="visOption.text"
+        hide-details
+        @change="() => updateVisualizationsShown(visOption)"
+      />
+      <div
+        v-if="showOptions"
+        class="px-5"
+      >
+        <v-checkbox
+          v-for="available in visOption.availableOptions"
+          :key="available.value"
+          :label="available.text"
+          dense
+          hide-details
+          class="my-0 py-0"
+          @change="() => updateOption(visOption, available.value)"
+        />
+      </div>
+    </div>
+  </v-list-item-content>
+</template>
+
+<script lang="ts">
+import { Dataset, Pin, VisOption } from '@/generatedTypes/AtlascopeTypes';
+import visualize, { visualizeDetectedStructures } from '@/utilities/analyticsVis';
+import {
+  computed, defineComponent, watch,
+} from '@vue/composition-api';
+import store from '../store';
+
+export default defineComponent({
+  props: {
+    pin: {
+      type: Object,
```

Comment:
Unfortunately not, because typescript types aren't sufficient for dictionary values, so we need to use a builtin. else we get this error: `'Pin' only refers to a type, but is being used as a value here.`

Reply:
Indeed, `Pin` in this context is a TypeScript `interface`, and what is needed here is a JavaScript value that fulfills that interface. If you renamed `Pin` (the `interface`) to `PinInterface`, and created a function named `Pin` that yields a value of type `PinInterface`, you could perhaps use `Pin` here in place of `Object`. Not sure about this, though--I'd have to try it and see what the typechecker thinks.